src/core-asyncpipeline/AsyncPipeline/AsyncResultFlow/Map/Map.Task.cs

```diff
@@ -0,0 +1,48 @@
+#nullable enable
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System
+{
+    partial struct AsyncResultFlow<TSuccess, TFailure>
+    {
+        public AsyncResultFlow<TResultSuccess, TResultFailure> Map<TResultSuccess, TResultFailure>(
+            Func<TSuccess, CancellationToken, Task<TResultSuccess>> mapSuccessAsync,
+            Func<TFailure, TResultFailure> mapFailure)
+            where TResultFailure : struct
+            =>
+            InnerMap(
+                mapSuccessAsync ?? throw new ArgumentNullException(nameof(mapSuccessAsync)),
+                mapFailure ?? throw new ArgumentNullException(nameof(mapFailure)));
+
+        public AsyncResultFlow<TResultSuccess, TResultFailure> Map<TResultSuccess, TResultFailure>(
+            Func<TSuccess, CancellationToken, Task<TResultSuccess>> mapSuccessAsync,
+            Func<TFailure, CancellationToken, Task<TResultFailure>> mapFailureAsync)
+            where TResultFailure : struct
+            =>
+            InnerMap(
+                mapSuccessAsync ?? throw new ArgumentNullException(nameof(mapSuccessAsync)),
+                mapFailureAsync ?? throw new ArgumentNullException(nameof(mapFailureAsync)));
+
+        private AsyncResultFlow<TResultSuccess, TResultFailure> InnerMap<TResultSuccess, TResultFailure>(
+            Func<TSuccess, CancellationToken, Task<TResultSuccess>> mapSuccessAsync,
+            Func<TFailure, TResultFailure> mapFailure)
+            where TResultFailure : struct
+            =>
+            InnerPipe(
+                (r, t) => r.MapAsync(
+                    s => mapSuccessAsync.Invoke(s, t),
+                    f => f.Pipe(mapFailure).Pipe(Task.FromResult)));
```

Comment:
Let's get AsyncPipeline independent from Primitives dependency.
Only the Pipe extension from the Primitives is used several times.

We can replace the Pipe by general C# method call kind of `Task.FromResult(mapFailure.Invoke(f))` or implement something like to keep the multistep pipelines:
```cs
internal static class InternalPipelineExtensions
{
    // The pipe arg not null validation is skipped as the internal extension is expected to be called in the correct context
    internal static TResult Pipe<T, TResult>(this T value, Func<T, TResult> pipe)
        =>
        pipe.Invoke(value);
}
```

Reply:
Done